tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh

```diff
@@ -219,14 +219,14 @@ i=1
 fs=$TESTPOOL/$TESTFS1
 log_must $ZFS create $fs
 log_must $ZFS snapshot $fs@snap
-while((i < 7)); do
+while((i < 21)); do
 	log_must $ZFS clone $fs@snap $fs/$TESTCLONE$(python -c 'print "x" * 200').$i
 	((i=i+1))
 	((j=j+200))
 done
 clone_list=$($ZFS list -o clones $fs@snap)
 char_count=$($ECHO "$clone_list" | $TAIL -1 | wc | $AWK '{print $3}')
-[[ $char_count -eq 1024 ]] || \
+[[ $char_count -eq 4096 ]] || \
```

Comment:
Just to be clear, this was caused by ZFS_MAXPROPLEN=1024 on illumos and ZFS_MAXPROPLEN=4096 on Linux.  Which is because MAXPATHLEN is defined differently.  I'd suggest reworking this slightly to use a variable to make this clear.  Something like:

```
if is_linux; then
        ZFS_MAXPROPLEN=4096        
else
        ZFS_MAXPROPLEN=1024
fi
```

Then use ZFS_MAXPROPLEN to calculate the number of iterations for the loop, say `$(( $ZFS_MAXPROPLEN/200+1 ))`.